news.js

```diff
@@ -0,0 +1,214 @@
+//variable used to store the complete data of all news channels
+let news;
+window.onload = () => {
+	console.log('onload');
+	news = new News();
+};
+class News {
+	constructor() {
+		this.fullData = '';
+		this.data1 = [];
+		this.email = [];
+		this.displayPage();
+	}
+
+	//Function to display the selected news category
+	selectCategoryNews = () => {
+		let selectedCategory = document.getElementById('sel-category').value;
+		let display = ' ';
+		let indexDisplay = [];
+		let data = this.data1;
+
+		let selectedData = data.filter(function(e, index) {
+			if (selectedCategory === e.source.name) indexDisplay.push(index);
+			return selectedCategory === e.source.name;
+		});
+
+		let popupIndex = 0;
+		selectedData.forEach((currentItem, index) => {
+			console.log(currentItem.urlToImage);
+			display += `<div class='content' id='content'>
+					<div class='content__sub' id='content__display'>
+					<img src='${currentItem.urlToImage}' class='content__img' ></img>
+					<h3 class='content__modifier content__head'>
+					${currentItem.title}
+					</h3> 
+					<p class='content__modifier content__date'>
+					${currentItem.publishedAt}
+					</p> 
+					<p class='content__modifier content__matter'>
+					${currentItem.description}
+					</p>
+					<a href='#' class='content__modifier btn btn--pink' id='myBtn${index}'>Continue Reading</a>
+					</div><hr></div>`;
+		});
+
+		document.getElementById('displaynews').innerHTML = display;
+		for (let index = 0; index < selectedData.length; index++) {
+			console.log(document.getElementById(`myBtn${index}`).innerHTML);
+			document.getElementById(`myBtn${index}`).addEventListener('click', function() {
+				console.log(data);
+				showpopup(indexDisplay[index], data);
+			});
+		}
+	};
+
+	//On Load Function Call
+
+	displayPage = () => {
+		this.header();
+		this.footer();
+		this.displayForm();
+		this.displayNews();
+	};
+
+	header = () => {
+		let html = `<h1 class="header__heading">NEWSFEED</h1><p class="header__caption">Yet another newsfeed</p>
+				<input type='text' id='subscribe' class='form__subscribe-textbox' placeholder='Email Address'/>
+				<button class='form__subscribe-button' id='subBtn'>Subscribe</button>`;
+		document.getElementById('header').innerHTML = html;
+		document.getElementById('subBtn').addEventListener('click', validate);
+	};
+
+	footer = () => {
+		let html = `<p class="footer__copyright">&copy; NewsFeed 2019</p>`;
+		document.getElementById('footer').innerHTML = html;
+	};
+
+	displayForm = () => {
+		let url = 'https://newsapi.org/v2/top-headlines?' + 'country=us&' + 'apiKey=9fdb04ee4078412b82f9dd7f760464f8';
+		let req = new Request(url);
+		fetch(req).then((res) => res.json()).then((data) => {
+			this.data1 = data.articles;
```

Comment:
Move the inside of success response to another function. Ideally any function should have a length of maximum 30-50 lines